sqlglot/generator.py

```diff
@@ -38,8 +38,18 @@ class Generator:
     """
 
     TRANSFORMS = {
+        exp.AnonymousProperty: lambda self, e: self.property_sql(e),
+        exp.AutoIncrementProperty: lambda self, e: f"AUTO_INCREMENT={e.text('value')}",
+        exp.CharacterSetProperty: lambda self, e: f"{'DEFAULT ' if e.args['default'] else ''}CHARACTER SET={e.text('value')}",
+        exp.CollateProperty: lambda self, e: f"COLLATE={e.text('value')}",
         exp.DateAdd: lambda self, e: f"DATE_ADD({self.sql(e, 'this')}, {self.sql(e, 'expression')}, {self.sql(e, 'unit')})",
         exp.DateDiff: lambda self, e: f"DATE_DIFF({self.sql(e, 'this')}, {self.sql(e, 'expression')})",
+        exp.EngineProperty: lambda self, e: f"ENGINE={e.text('value')}",
+        exp.FileFormatProperty: lambda self, e: f"format = '{e.text('value')}'",
```

Comment:
looks like your spacing is not consistent? also should it be caps FORMAT

Reply:
This has to do with the identity tests. They expect root expressions to have the space and lowercase while WITH expressions to have no space and uppercase. Do we want to change the identity tests to reflect a more consistent approach?

Reply:
you can fix the tests yes